### PR TITLE
Feature/tao 5040/proctor time countdown

### PR DESCRIPTION
**Pull request by tikhanovichA**
Added proctor time countdown. Fixed time mismatch on Proctor screen and test.
https://oat-sa.atlassian.net/browse/TAO-5040
https://oat-sa.atlassian.net/browse/TAO-5037

### CODE REVIEW

**Comment by krampstudio** [comment]
Please add any backend developer for the backend code review

File: views/js/controller/Delivery/monitoring.js [hunk 1]
```diff
@@ -516,6 +518,96 @@ define([
                 }
 
                 /**
+                 * Formats a timer
+                 * @param {Number} totalSeconds
+                 * @returns {String}
+                 */
+                function formatTime(totalSeconds) {
```
Comment (krampstudio) [inline views/js/controller/Delivery/monitoring.js]: You have access to the `moment` library in this module : http://momentjs.com/ please use it's formatting features

File: views/js/controller/Delivery/monitoring.js [hunk 2]
```diff
@@ -516,6 +518,96 @@ define([
...
+                        minutes = "0" + minutes;
+                    }
+                    if (seconds < 10) {
+                        seconds = "0" + seconds;
+                    }
+
+                    return hours + ':' + minutes + ':' + seconds;
+                }
+
+                function updateTimer(rows) {
```
Comment (krampstudio) [inline views/js/controller/Delivery/monitoring.js]: You shouldn't code a new timer countdown but reuse the one available.
See `core/polling` and `core/timer`. The  [timer plugin](https://github.com/oat-sa/extension-tao-testqti/blob/master/views/js/runner/plugins/controls/timer/timer.js#L404) has a good example.

File: model/implementation/DeliveryExecutionStateService.php [hunk 3]
```diff
@@ -60,6 +60,18 @@ class DeliveryExecutionStateService extends AbstractStateService implements \oat
     private $testSessionService;
 
     /**
+     * Computes the state of the delivery and returns one of the extended state code
+     *
+     * @param DeliveryExecution $deliveryExecution
+     * @return null|string
+     * @throws \common_Exception
+     */
+    public function getState(DeliveryExecution $deliveryExecution)
```
Comment (boajer) [inline model/implementation/DeliveryExecutionStateService.php]: What is the reason to introduce this public method?  Have you check other possible implementations if any?

File: model/monitorCache/update/TestUpdate.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+use oat\taoDelivery\model\execution\ServiceProxy;
+use oat\taoProctoring\model\monitorCache\DeliveryMonitoringService;
+use oat\taoQtiTest\models\event\QtiTestChangeEvent;
+use oat\oatbox\service\ServiceManager;
+use qtism\runtime\tests\AssessmentTestSessionState;
+
+/**
+ *
+ * @package oat\taoProctoring
+ * @author Aleh Hutnikau <hutnikau@1pt.com>
```
Comment (boajer) [inline model/monitorCache/update/TestUpdate.php]: Wrong author?

File: model/monitorCache/update/TestUpdate.php [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015 (original work) Open Assessment Technologies SA;
```
Comment (boajer) [inline model/monitorCache/update/TestUpdate.php]: 2015?

**Pull request merged**